mindmeld/active_learning/plot_manager.py

```diff
@@ -0,0 +1,471 @@
+            experiment_folder=self.experiment_dir_path
+        )
+        with open(selected_queries_json_path, "r") as infile:
+            return json.load(infile)
+
+    def queries_json_data_has_data(self) -> Dict:
+        """Checks whether queries.json is empty {}.
+        Returns:
+            has_data (bool): Whether queries.json has data.
+        """
+        return {} != self.get_queries_json_data()
+
+    # Plotting Meta Functions
+    def generate_plots(self):
+        """Entry point for generating plots as per the specifications provided in the config."""
+        logger.info("Starting to generate plots")
+        plot_functions = [
+            "plot_single_epoch",
+            "plot_avg_across_epochs",
+            "plot_all_epochs",
+        ]
+        for plot_function in plot_functions:
+            logger.info("Plotting: %s", plot_function)
+            plot_function = getattr(PlotManager, plot_function)
+            self.plotting_wrapper(plot_function)
+        if self.queries_json_data_has_data():
+            logger.info("Plotting: plot_stacked_bar")
+            self.plot_stacked_bar()
+
+    def plotting_wrapper(
+        self,
+        function,
+        plot_domain: bool = True,
+        plot_intents: bool = True,
+        plot_entities: bool = False,
+    ):
+        """Plotting wrapper functions for plots that use data from accuracies.json
+        Args:
+            function (generator): plotting function.
+            plot_domain (bool): Whether to generate plots at the domain level.
+            plot_intents (bool): Whether to generate plots at the intent level.
+            plot_entities (bool): Whether to generate plots at the entity level.
+        """
+        if plot_domain:
+            function(self, y_keys=["overall"])
+            for domain in self.get_domain_list():
+                function(self, y_keys=[domain, "overall"])
+                if plot_intents:
+                    for intent in self.get_intent_list(domain):
+                        function(self, y_keys=[domain, intent, "overall"])
+                        if plot_entities:
+                            y_keys = [domain, intent, "entities", "overall"]
+                            function(self, y_keys=y_keys)
+                            for entity in self.get_entity_list(domain, intent):
+                                y_keys = [domain, intent, "entities", entity]
+                                function(self, y_keys=y_keys)
+
+    # Helper Methods
+    @staticmethod
+    def get_nested(data_dict: Dict, keys: List):
+        """Get data from a nested dictionary using a series of keys.
+        Args:
+            data_dict (dict): Dictionary containing data
+            keys (list): List of keys used to index the given dictionary.
+
+        Returns:
+            data_dict (dict): Dictionary containing the nested data.
+        """
+        for key in keys:
+            data_dict = data_dict[key]
```

Comment:
this operation is weird, we are reassigning a dict for all keys. So isnt it just the `data_dict = data_dict[keys[-1]]` ?

Reply:
The purpose of this function is to filter data from a nested dictionary by selecting from a set of “selected_keys”. I have updated the documentation and variable names here to clarify this behavior.